sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/postgres/PostgresInt2Tests.java

```diff
@@ -0,0 +1,397 @@
+        );
+    }
+
+    @Test @Ignore("unimplemented operator")
+    public void testLeftShift() {
+        this.qs(
+        "SELECT (-1::INT2 << 15)::TEXT;\n" +
+                "  text  \n" +
+                "--------\n" +
+                " -32768\n" +
+                "(1 row)\n" +
+                "\n" +
+                "SELECT ((-1::int2<<15)+1::int2)::text;\n" +
+                "  text  \n" +
+                "--------\n" +
+                " -32767\n" +
+                "(1 row)"
+        );
+    }
+
+    // Ignored because this fails in Postgres but here we get:
+    @Test @Ignore
+    public void testSelectOverflow() {
+        String error_message = "INT2 out of range";
+        // We get:
+        // L: (Some(-32767), Some(2))x1 --> wraps around
+        // L: (Some(-1234), Some(-2468))x1
+        // L: (Some(0), Some(0))x1
+        // L: (Some(1234), Some(2468))x1
+        // L: (Some(32767), Some(-2))x1 --> wraps around
+        // Taken from: https://github.com/postgres/postgres/blob/c161ab74f76af8e0f3c6b349438525ad9575683b/src/test/regress/expected/int2.out#L202C1-L203C30
+        this.shouldFail("SELECT i.f1, i.f1 * 2::INT2 AS x FROM INT2_TBL i", error_message);
+
+        // We get:
+        // L: (Some(-32767), Some(-32765))x1
+        // L: (Some(-1234), Some(-1232))x1
+        // L: (Some(0), Some(2))x1
+        // L: (Some(1234), Some(1236))x1
+        // L: (Some(32767), Some(-32767))x1 --> wraps around
+        // https://github.com/postgres/postgres/blob/c161ab74f76af8e0f3c6b349438525ad9575683b/src/test/regress/expected/int2.out#L223
+        this.shouldFail( "SELECT i.f1, i.f1 + '2'::INT2 AS x FROM INT2_TBL i", error_message);
+
+        // Similarly,
+        this.shouldFail("SELECT i.f1, i.f1 - '2'::INT2 AS x FROM INT2_TBL i;", error_message);
+    }
+
+    // This passes for the Calcite version but fails for the run time version
+    // error [E0080]:
+    //  |
+    //2 | println!("{}", -32768i16 % -1i16);
+    //  |                ^^^^^^^^^^^^^^^^^ attempt to compute `i16::MIN % -1_i16`, which would overflow
+    //  |
+    //  = note: `#[deny(unconditional_panic)]` on by default
+    //  |
+    //2 | println!("{}", -32768i16 % -1i16);
+    //  |                ^^^^^^^^^^^^^^^^^ overflow in signed remainder (dividing MIN by -1)
+    @Test @Ignore
+    public void testINT2MINOverflow() {
+        this.q(
+                "SELECT (-32768)::int2 % (-1)::int2 as x;\n" +
+                        " x \n" +
+                        "---\n" +
+                        "  0"
+        );
+    }
+
+    @Test @Ignore
+    public void testINT2MINOverflowError() {
+        String error = "INT2 out of range";
+
+        // This fails in Postgres, but we get: `-32768`
+        this.shouldFail("SELECT (-32768)::int2 * (-1)::int2", error);
+
+        // This panics in run time
+        this.shouldFail("SELECT (-32768)::int2 / (-1)::int2", error);
+    }
+
+    @Test @Ignore("underscores not supported yet")
```

Comment:
I don't think we plan to support them. Don't waste your time creating tests like this. You can just leave a large comment in the file with the SQL and the mention that we don't support this.